README
==========================================================================

Université de Strasbourg - Direction Informatique
Auteur : Guilhem BORGHESI
Création : Février 2008

borghesi@unistra.fr

Ce logiciel est régi par la licence CeCILL-B soumise au droit français et
respectant les principes de diffusion des logiciels libres. Vous pouvez
utiliser, modifier et/ou redistribuer ce programme sous les conditions
de la licence CeCILL-B telle que diffusée par le CEA, le CNRS et l'INRIA 
sur le site "http://www.cecill.info".

Le fait que vous puissiez accéder à cet en-tête signifie que vous avez 
pris connaissance de la licence CeCILL-B, et que vous en avez accepté les
termes. Vous pouvez trouver une copie de la licence dans le fichier LICENCE.

==========================================================================

Université de Strasbourg - Direction Informatique
Author : Guilhem BORGHESI
Creation : Feb 2008

borghesi@unistra.fr

This software is governed by the CeCILL-B license under French law and
abiding by the rules of distribution of free software. You can  use, 
modify and/ or redistribute the software under the terms of the CeCILL-B
license as circulated by CEA, CNRS and INRIA at the following URL
"http://www.cecill.info". 

The fact that you are presently reading this means that you have had
knowledge of the CeCILL-B license and that you accept its terms. You can
find a copy of this license in the file LICENSE.

==========================================================================

=============================================================================
Fichiers de l'application
=============================================================================

index.php
	La page d'accueil de STUdS
studs.php
	La page de présentation de sondage
adminstuds.php
	La page d'administration réservée à l'auteur du sondage
infos_sondage.php
	La page (1/2) de création de sondage récupérant les informations générales
choix_date.php
	La page de création (2/2) pour un sondage pour déterminer une date
choix_autre.php
	La page de création (2/2) pour un sondage sur un sujet quelconque
creation_sondage.php
	Le fichier qui récupérent les informations des pages précédentes pour procéder à l'insertion du nouveau sondage dans la base PostgreSQL
style.css
	Le fichier CSS de style pour toute l'application
variables.php
	Le fichier contenant les variables à changer en fonction de la machine locale
fonctions.php
	Le fichier contenant quelques fonctions récurrentes de l'application
i18n.php
	Le fichier contenant quelques fonctions récurrentes de l'application relatives à l'internationalisation
README
	Ce fichier
INSTALL
	Le fichier contenant les informations d'installation sur l'application
CHANGELOG
	Le fichier contenant toutes les modifications de l'application entre les différentes versions
contacts.php
	La page permettant aux usagers de poser une question à l'administrateur de l'application
apropos.php
	La page expliquant les détails techniques relatifs à l'application et les dernieres modifications et celles à venir sur l'application
bandeaux.php
	Le fichier contenant tous les bandeaux des pages PHP de l'application
favicon.ico
	L'icone de favoris de l'application
sources.php
	La page qui propose les sources de l'application
exportics.php
	Le fichier d'export de la meilleure date au format iCAL (fichier .ICS)
exportcsv.php
	Le fichier d'export de tous le tableau des participants avec leurs réponses dans un tableur (format .CSV)
exportpdf.php
	Le fichier d'export de la lettre de convocation que le créateur du sondage pourra envoyer aux participants (format .PDF)

admin/
	Le répertoire réservé à l'administrateur de l'application
admin/.htaccess
	Le fichier gérant les droits restreints du répertoire ADMIN
admin/.htpasswd
	Le fichier contenant les passwd des logins ayant accès au répertoire ADMIN
admin/index.php
	La page présentant tous les sondages actuellement dans la base à l'administrateur
admin/log_studs.txt
	Le fichier contenant un historique de toutes les creations/suppressions de sondage dans la base

errors/
	Le répertoire contenant toutes les pages d'erreurs
errors/error-forbidden.php
	La page qui indique dans la charte graphique de l'application l'erreur "501 forbidden"
errors/maintenance.php
	La page qui indique que l'application est en maintenance temporaire

export/
	Le répertoire qui contient tous les exports ICS

iCalcreator/
	Le répertoire qui contient les librairies d'export en iCal

php2pdf/
	Le répertoire qui contient les librairies d'export en PDF

scripts/
	Le répertoire qui contient tous les scripts de l'application
	
sources/
	Le répertoire qui contient les sources de l'application disponible sur la page sources.php

locale/
	Le répertoire qui contient les fichiers de traduction modifiables (.po) et compilés (.mo)
	au format gettext

=============================================================================
	Validations des pages
=============================================================================

Toutes les pages de STUdS sont validées HTML 4.01 Strict.
La CSS de STUdS est validée CSS 2.1.

=============================================================================
Technologies utilisées
=============================================================================

- PHP, php-fpdf, php-adodb, php-gettext
- PostgreSQL, mysql
- Apache
- Subversion
- iCalcreator
- POedit
- Icônes : Deleket (http://deleket.deviantart.com/) et DryIcons (http://dryicons.com)

=============================================================================
Compatibilités des navigateurs
=============================================================================

- Firefox 
- Opera 
- Konqueror
- Links 
- Safari
- IE 
	
-----------------
Janvier 2008
Guilhem BORGHESI
Université de Strasbourg

Mai 2010
Raphaël DROZ, raphael.droz@gmail.com



CHANGELOG
Les dernières améliorations de STUdS
	Changelog version 0.6.6 (XXX 2011) :
	- internationalisation avec gettext
	- abstraction de la base de données avec ADOdb
	- support de mysql (fichier d'initialisation disponible)
	- meilleure compatibilité avec le mode strict de PHP
	- factorisation de code et de CSS
	- moins de boutons de formulaire, plus de liens <a href>

	Changelog version 0.6.5 (juin 2010) :
	- Changement de deux icones dans la creation d'un sondage.

	Changelog version 0.6.4 (mars 2010) :
	- Corrections de bug

	Changelog version 0.6.3 (janvier 2010) :
	- Corrections de bug

	Changelog version 0.6.2 (novembre 2009) :
	- Correction dans l'affichage des bandeaux,
	- Modification de la partie "A propos",
	- Préparation à l'authentification,
	- De UdSification de l'application dans certains fichiers.

	Changelog version 0.6.1 (octobre 2009) :
	- Corrections d'erreurs dans les traductions et d'oublis de traduction dans certaines pages.

	Changelog version 0.6 (août 2009) :
	- Mise sous la licence CeCILL-B du code source de STUdS,
	- Passage de STUdS en encodage UTF8,
	- Ajout des icones des menus dans toutes les pages et non pas seulement sur la page d'accueil,
	- Correction d'un bug lors du rajout d'une colonne dans l'interface d'administration des sondages.

	Changelog version 0.5 (février 2009) :
	- Traduction de STUdS en anglais, allemand et espagnol,
	- Changement de la CSS avec ajout du logo de l'Université de Strasbourg,
	- Possibilité d'ajouter un commentaire pour les sondés.
	
	Changelog version 0.4 (janvier 2009) :
	- Possibilité de faire un export PDF pour envoyer la lettre de convocation à la date de réunion,
	- Possibilité de rajouter des colonnes dans la partie administration de sondage,
	- Correction de bugs d'affichage avec les caractères ' et " .

	Changelog version 0.3 (novembre 2008) :
	- Possibilité de faire un export CSV pour exploiter le sondage dans un tableur,
	- Mise en place d'un repository Subversion pour partager les nouvelles versions de STUdS,
	- Amélioration de la CSS pour un meilleur affichage,
	- Modification du code source pour le rendre portable vers une autre machine.
	
	Changelog version 0.2 (novembre 2008) :
	- Lors de la création d'un sondage DATE, classement des dates par ordre croissant,
	- Lors de la création d'un sondage DATE, accepter les horaires au format "8h" ou "8H",
	- Lors de la création d'un sondage DATE, possibilité de copier des horaires entre les dates,
	- Lors d'une modification de ligne, cocher les cases initialement choisies et non pas des cases vides,
	- Changement du format d'affichage des dates pour un formatage type : "Mardi 13/06",
	- Meilleure visualisation des choix les plus votés,
	- Possibilité pour l'administrateur du sondage de choisir de recevoir un mail d'alerte à chaque participation d'un sondé,
	- Remplacement des boutons de formulaire par des images moins austères,
	- Correction de quelques petits bugs d'affichage,
	- Possibilité de rajouter des cases supplémentaires lors de la création d'un sondage AUTRE,
	- Possibilité de rajouter des cases supplémentaires lors de la création d'un sondage DATE.


apropos.php
<?php
//==========================================================================
//
//Université de Strasbourg - Direction Informatique
//Auteur : Guilhem BORGHESI
//Création : Février 2008
//
//borghesi@unistra.fr
//
//Ce logiciel est régi par la licence CeCILL-B soumise au droit français et
//respectant les principes de diffusion des logiciels libres. Vous pouvez
//utiliser, modifier et/ou redistribuer ce programme sous les conditions
//de la licence CeCILL-B telle que diffusée par le CEA, le CNRS et l'INRIA 
//sur le site "http://www.cecill.info".
//
//Le fait que vous puissiez accéder à cet en-tête signifie que vous avez 
//pris connaissance de la licence CeCILL-B, et que vous en avez accepté les
//termes. Vous pouvez trouver une copie de la licence dans le fichier LICENCE.
//
//==========================================================================
//
//Université de Strasbourg - Direction Informatique
//Author : Guilhem BORGHESI
//Creation : Feb 2008
//
//borghesi@unistra.fr
//
//This software is governed by the CeCILL-B license under French law and
//abiding by the rules of distribution of free software. You can  use, 
//modify and/ or redistribute the software under the terms of the CeCILL-B
//license as circulated by CEA, CNRS and INRIA at the following URL
//"http://www.cecill.info". 
//
//The fact that you are presently reading this means that you have had
//knowledge of the CeCILL-B license and that you accept its terms. You can
//find a copy of this license in the file LICENSE.
//
//==========================================================================

session_start();

include_once('variables.php');
include_once( 'i18n.php' );
if (file_exists('bandeaux_local.php'))
	include_once('bandeaux_local.php');
else
	include_once('bandeaux.php');

	//affichage de la page
	echo '<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN">'."\n";
	echo '<html>'."\n";
	echo '<head>'."\n";
	echo '<meta http-equiv="Content-Type" content="text/html; charset=utf-8">'."\n";
	echo '<title>'.getenv('NOMAPPLICATION').'</title>'."\n";
	echo '<link rel="stylesheet" type="text/css" href="style.css">'."\n";
	echo '</head>'."\n";
	echo '<body>'."\n";

    //debut du formulaire
    echo '<form name=formulaire action="apropos.php" method="POST">'."\n";

	//bandeaux de tete
	logo();
	bandeau_tete();
	bandeau_titre_apropos();
	sous_bandeau();

	//blablabla
	echo '<div class=corps>'."\n";

	if (getenv('NOMAPPLICATION')!="STUdS !"){
		echo '<b>Application d\'origine</b><br><br>'."\n";
		echo 'L\'application '.getenv('NOMAPPLICATION').' est une instance du logiciel <b><a href ="http://studs.u-strasbg.fr">STUdS !</a></b> d&eacute;velopp&eacute; &agrave; l\'Universit&eacute; de Strasbourg depuis 2008.<br><br>'."\n";
	}

	echo '<b>Licence Logicielle de '.getenv('NOMAPPLICATION').'</b><br><br>'."\n";
	echo getenv('NOMAPPLICATION').' est plac&eacute; sous la licence logicielle libre <a href="http://www.cecill.info/licences.fr.html">CeCILL-B</a>.<br><br>'."\n";

	echo '<b>Technologies utilis&eacute;es</b><br><br>'."\n";
	echo '- <a href="http://www.php.net/">PHP</a><br>'."\n";
	echo '- <a href="http://www.postgresql.org/">PostgreSQL</a><br>'."\n";
	echo '- <a href="http://www.apache.org/">Apache</a><br>'."\n";
	echo '- <a href="http://subversion.tigris.org/">Subversion</a><br>'."\n";
	echo '- <a href="http://www.kigkonsult.se/iCalcreator/">iCalcreator</a><br>'."\n";
	echo '- <a href="http://www.fpdf.org/">FPDF</a><br>'."\n";
	echo '- Ic&ocirc;nes : <a href="http://deleket.deviantart.com/">Deleket</a>, <a href ="http://pixel-mixer.com">PixelMixer</a> et <a href="http://dryicons.com">DryIcons</a><br><br>'."\n";
	
	echo '<b>Compatibilit&eacute;s des navigateurs</b><br><br>'."\n";
	echo '- <a href="http://www.mozilla.com/firefox/">Firefox</a><br>'."\n";
	echo '- <a href="http://www.opera.com/">Op&eacute;ra</a><br>'."\n";
	echo '- <a href="http://www.konqueror.org/">Konqueror</a><br>'."\n";
	echo '- <a href="http://www.jikos.cz/~mikulas/links/">Links</a><br>'."\n";
	echo '- <a href="http://www.apple.com/fr/safari/">Safari</a><br>'."\n";
	echo '- <a href="http://www.mozilla.com/firefox/">IE</a><br><br>'."\n";

	echo '<b>Validations des pages</b><br><br>'."\n";
	echo '- Toutes les pages disposent de la validation HTML 4.01 Strict du W3C. <br>- La CSS dispose de la validation CSS 2.1 du W3C.'."\n";
 	echo '<p>'."\n"; 
	echo '<img src="http://www.w3.org/Icons/valid-html401-blue" alt="Valid HTML 4.01 Strict" height="31" width="88"><img style="border:0;width:88px;height:31px" src="http://jigsaw.w3.org/css-validator/images/vcss-blue" alt="CSS Valide !">'."\n";
 	echo'</p>'."\n"; 

	echo '<b>Propositions am&eacute;liorations de '.getenv('NOMAPPLICATION').'</b><br><br>'."\n";
	
	echo 'Si quelquechose venait &agrave; vous manquer, vous pouvez nous en faire part via le <a href="contacts.php">formulaire en ligne</a>. <br>'."\n";

	echo 'Les derni&egrave;res am&eacute;liorations de '.getenv('NOMAPPLICATION').' sont visibles dans le fichier <a href="CHANGELOG">CHANGELOG</a>.<br><br>'."\n";
	
	echo '<b>Remerciements</b><br><br>'."\n";
	echo 'Pour leurs contributions techniques ou ergonomiques : Guy, Christophe, Julien, Pierre, Romaric, Matthieu, Catherine, Christine, Olivier, Emmanuel et Florence <br><br>'."\n";

	

	echo '</div>'."\n";
	
	sur_bandeau_pied_mobile();
	bandeau_pied_mobile();
	echo '</form>'."\n";
	echo '</body>'."\n";
	echo '</html>'."\n";

?>
